### ISSUE TITLE
Code inside ActiveRecord lifecycle callbacks unable to change locales

### ISSUE THREAD

**Issue by woahdae** [globalize/globalize#659]
[Globalize::ActiveRecord::InstanceMethods#save](https://github.com/globalize/globalize/blob/ef1e362f3144b2276d89f47407c67ffa3fc1f9f2/lib/globalize/active_record/instance_methods.rb#L145) wraps `ActiveRecord::Base#save`, and therefore all lifecycle methods, with an explicit set of `Globalize#locale` equal to the localization value at the point when `#save` was called. 

This is problematic because it makes calls to `I18n.with_locale` (and friends) in an after_save not change `Globalize.locale`, and therefore makes translated models not work in that context. My use-case is indexing translations in elasticsearch via an after_save callback, where I was surprised to find that only in `#after_save` my models were not responding to switching locales.

In other words, in any AR lifecycle callback, the following holds true so long as the model was saved while `I18n.locale == :en`:

    I18n.with_locale(:es) { Globalize.locale } #=> :en

[Here's a gist with the repro case](https://gist.github.com/woahdae/ec3a2445ade123882643478e0898a572).

Possibly related to #476 and/or #460

**Comment by woahdae** [comment]
Hi stalebot, it is stale because nobody responded (globalize doesn’t owe me anything, nbd) but it’s still an issue.

**Comment by parndt** [comment]
@woahdae indeed, because of the mountain of issues I've had to add the bot. Sorry that nobody replied. :slightly_smiling_face: do you have any interest (or time) to convert your gist into a pull request that adds those tests?

**Comment by brucebolt** [comment]
We have experienced the same issue in a `after_save` callback (which pushes the contents of the model, including translations, into another API).

Within our `with_locale` block, we have added `Globalize.locale = I18n.locale`, which has solved the problem.  Although it'd be great if there were some way of fixing this without needing this hack.
